tests/clpy_tests/opencl_tests/ultima_tests/utility.py

```diff
@@ -0,0 +1,58 @@
+import clpy
+
+import os
+import subprocess
+import tempfile
+import time
+
+
+class TempFile(object):
+    def __init__(self, filename, source):
+        self.fn = filename
+        self.s = source
+
+    def __enter__(self):
+        with open(self.fn, 'w') as f:
+            f.write(self.s)
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        os.remove(self.fn)
+
+
+def _exec_ultima(source, _clpy_header=''):
```

Comment:
The name should not start with `_` because it's not internal function.

Reply:
I fixed at 21783b4.